server/src/main/java/io/crate/planner/WhereClauseOptimizer.java

```diff
@@ -161,6 +161,9 @@ public static DetailedQuery optimize(EvaluatingNormalizer normalizer,
 
         EqualityExtractor eqExtractor = new EqualityExtractor(normalizer);
         List<List<Symbol>> pkValues = eqExtractor.extractExactMatches(pkCols, query, txnCtx);
+        if (pkValues == null) {
+            pkValues = eqExtractor.extractExactMatches(List.of(DocSysColumns.ID), query, txnCtx);
```

Comment:
I wonder if there's a way to speed this up by re-using parts of the first pass?

If I remember right the EqualityExtractor tends to show up in planner flamegraphs when profiling

Reply:
On the top of my head, and since to return `pkValues` here the query must match all of the pk columns - so we cannot just add `_id` to the `table.primaryKey()`, I could try to pass a List<List<ColumnIdent>> and have an extra iteration when matching query symbols. wdyt?

Reply:
>  I could try to pass a List<List> and have an extra iteration when matching query symbols. wdyt?

Sounds good. I'm not sure if it is possible, but if we can re-use some of the "content preparation" that's happening within the EqualityExtractor it would be nice.